### PR TITLE
[stacks-core] Credential loading and generation

### PR DESCRIPTION
**Pull request by stjepangolemac**
Up until now we've been directly constructing Stacks addresses from generated private keys. The rest of the ecosystem derives keys from the seed private key and we're now bringing `stacks-core` up to date.

Also, we need this for further sBTC development — to be on par with @radicleart's work with the bridge.

- [x] Proper key derivation with Stacks derivation path
- [x] Generation of new credentials with `sbtc-cli`
- [x] Usage in Romeo (to be added as soon as I confirm credentials are okay with @radicleart)

#### Note

We now have the same Stacks addresses as Leather wallet for the same private key, ~~but for some reason BTC ones are different.~~ I'm looking into it but I'd appreciate any ideas you might have.

I've fixed this now.

<details>

<summary>Example credential generation</summary>

```json
{
  "credentials": {
    "0": {
      "bitcoin": {
        "p2pkh": {
          "address": "myqbAU6JidUuXy5iFRhSL6BWhRCRvLktWS",
          "private_key": "44b4a6a203b31492f7d38609bec7d3645a496b065c86cbaae2c7fffff469e90e",
          "public_key": "0286ff20801a7015f88a21f8502a9fc9e9a13f8ff18e16344df56b17f3a3b02a05",
          "wif": "cPtFq429jih9Raft1NjvcvgPy6hnnLmYGjyMzJfyWyNYSA1fe89N"
        },
        "p2tr": {
          "address": "tb1pt9jngkz0w00y3y50usuu4vvv6t35vm3m7n55jthdlh45v2fgpp0sw00nx3",
          "private_key": "2055509bcead433f9ae42ba8f16d9c2469c6b5b6976e38855c050de6964c2b0c",
          "public_key": "0207a75bf8dced2280e26eb78ba75e9ba91ebf70e726d2db79f7a5f73056fe3d0e",
          "wif": "cNfZ3UGwyn8jzJWjJ9tLtpXPoWvYKafVGCD5fMM62sAkZ2pk8PJM"
        },
        "p2wpkh": {
          "address": "tb1q3s73ycrxr8c289dy4qcj5f5vkyp8mpnnek3mf7",
          "private_key": "101334877bc2dad586fb4cd9cc2adea73ad0ede3ed3b246c13e46a5f5be7455a",
          "public_key": "032d485d07a322b56a7d5fa34aaac1db2c4ac08c155ca30fd977329b9e0b74cbee",
          "wif": "cN7x1xwZHoQnhgogz4E6Yfs1swZiY2MapmeMDkCXsjCfcwc5YjrL"
        }
      },
      "stacks": {
        "address": "ST37CVTVGHRBGV7VBV4EYC6EYWNTSSCTVF15N0CHN",
        "private_key": "520572ecd22beb180581dff9411c971e8f4b3560688d65a1001b7fc9e67fd8e1",
        "public_key": "03ad4315ce794790edbb01f661d5e713a1fb0c4359f22b4341618d10e3e49ed3b1",
        "wif": "cQL96DtqXDRRXXQmumxG8RUZFEQdHF3ZQWqxixjQoSc8eukwJUVf"
      }
    },
    "1": {
      "bitcoin": {
        "p2pkh": {
          "address": "n3s5iS77KqrBHkKdcK4weNzvom1AJHymkg",
          "private_key": "ec4b843f5fa95112e232abe4ac5712902b73e1828af65d0c41a39a963b21280b",
          "public_key": "031d40a8956e48457b7dc0a37a0b9ad248d224df5068131890b923997ae9d1a19c",
          "wif": "cVW2bB6xqBhzYSLDyfqxzomnjJDh67NwcUkMHPAsxiuptJnLoNS4"
        },
        "p2tr": {
          "address": "tb1p7ydpk30xn84kn5dmqdqdua8nn97qqytus0shhfu9k7xrfgptkwjs42en79",
          "private_key": "1a6b09af8bafcdd9c3e878433a5d367cf632e4689bd1d1efbee4e07e3f39dd5e",
          "public_key": "0378141cf2438d9f41e7d1e3e43bfc777ef4a3c25ef6956226a3141a44ae36ecfe",
          "wif": "cNU49J9LH2svw4NmDod7HpmqK5QtPxBcqJXW66TrKwQfh4MgqpiE"
        },
        "p2wpkh": {
          "address": "tb1q95sqln2nqrfc8j39heqtq9rzwh9a8ccxf3kzep",
          "private_key": "05eab05171ae97da0a54d20440689af56444fbdc5a7a874daf96f100299c81fd",
          "public_key": "02c1c6fb17a9b5d4acc308acdbfb1eb695b7601df30f18963cc5f8f450ca79a5b2",
          "wif": "cMnCjGCYPebckNfcDqLrkTo5mkQgdL1wDS4BKVSTTHD4Qam5poG6"
        }
      },
      "stacks": {
        "address": "STRT1XNPQSFR0P5HT8EKPCD6VKDPFKR6TVY6ZQAD",
        "private_key": "c97314c1e5d2c4e7a120c541db47e08a0f20d2017b5dc7dc2f5fc3b9f7d8b899",
        "public_key": "02ce2f410be7497cd194a5e49ec386b5c6986b0ab937bc0b107c2ad74048f7b12a",
        "wif": "cULHxd47itezXoYUYn3WMRH6v6xyZ27JhJ95GUeWjvVUCdr2XmJo"
      }
    },
    "2": {
      "bitcoin": {
        "p2pkh": {
          "address": "mo2JVWkR5PDHjCSgchmNUR91Ve6skCi8sH",
          "private_key": "002459c6183000214248931a76d2f7fdc8b1898466c753ff9324adf4a78f7351",
          "public_key": "0348c52603ec8429d1c80cf89c9c84f23dd3f31ec5d18d52a3a3966a6a9e4b6812",
          "wif": "cMayf6pahpeM6xqBpi2FmrSuUBnJbkBirhLVpPBzkPwioDK3qz8J"
        },
        "p2tr": {
          "address": "tb1pqa087vhpdqvcff58lsg338vjuwe4gqwtu4dlndgv5e6ppzfwtupqz0a3pu",
          "private_key": "242447f2f38e0959fc5f8977604f6081975bc803718588961fb5bc4b5af4b9ec",
          "public_key": "024e0a998887ae34ad1b35e63e6c2c20698434f33a52762862d8304d35b7973352",
          "wif": "cNnxRg1gUTMpphHNNxD8Ma9yoJtDBprCXhPSgnA53Qy5oNkT2utr"
        },
        "p2wpkh": {
          "address": "tb1q66vgusac2t8xgue3p6h9zw4f2s85lrdwr7fuus",
          "private_key": "53da01b10912adaba775667c2a1301363f3e131b144b75eff077082f18000319",
          "public_key": "029328d76e678942dc5cffd5f9263c7599dd486b38eb1417dd357e0bb751b02ed0",
          "wif": "cQPhSuAcR8q5D4FprgoCEMLwoyAoCZTw1tMG5szHVshBXeYcAWvc"
        }
      },
      "stacks": {
        "address": "ST2ZC0QM0CZPHMQMVMCPQ2V5TZEZCYH26Q2H8EK4V",
        "private_key": "d35089a207d86b74f5ebc46db3c495371f108437133ccf9b0f80c369766a27be",
        "public_key": "036d91b6a1f38a4889e197a93677c7cde7e04d29f790bad550daff99b97221d574",
        "wif": "cUfUC28iqe9DW8gZMuPLKgv5AA7PiA7XwVctuQuNQEcr1mu2TeUi"
      }
    },
    "3": {
      "bitcoin": {
        "p2pkh": {
          "address": "myKh2zNC9L59Njn2EhtKuBPTc2eLySHcZ9",
          "private_key": "27f118c2923fda6e4c856889fc1ad18cf844f5d4dc0ef8ddd345dc0d419a08f9",
          "public_key": "03917c4ec468a1944d0199864963b88f29342813c86f3891348d89456293a277eb",
          "wif": "cNvLrwhvUchBFP9zzoL6RXLTYrvYHftTPQd71fTHLQycXQXYYiQy"
        },
        "p2tr": {
          "address": "tb1p694r2th77ye4kd6935p9jt0mn8x5v5tlxpsx0f24dtykj9544wssj9cntu",
          "private_key": "b657b8f8e040e41e9cfa2099d3ba3a5caca71ea72f762836cfb2cf601e32da65",
          "public_key": "03e9780d635ca1d47ee962fd1df2e503aa8c51f7d1784c46704497ea5f64e310df",
          "wif": "cTh9mS6dfrxwTACJpM1knHvYZ7iwcVAMATe4prZwz7ZxWs8CcY3e"
        },
        "p2wpkh": {
          "address": "tb1qsg6xfsrrnu7a7hc6pg2g53q6ld297cs4ly3mus",
          "private_key": "ff255ac47342cc5a09b31f4122b3870a9e1de016567f2753c88f858e7c7e02dc",
          "public_key": "025e978d94df95b69ac3eaaf4faccdb018425bbdb1397afa4a7d28fdad232c0032",
          "wif": "cW8fvjEQEhKLx64y796NXWMFVVsfAHGY5dvHGB5nR3fRCh7n4kn6"
        }
      },
      "stacks": {
        "address": "ST398P1QK5K0PZSJP4H0CNNJGXPVS0T3MFZV4RN6J",
        "private_key": "7a45f48cf96dd62bddc4017cf194262006db016f6f1f903412ea0d39e33e711a",
        "public_key": "02a11a15a46168e0cc3739cefcc169aaa2fbf98c0e63ecc9f4821476b8fdc4afc6",
        "wif": "cRgPH1QEUWGtdaqGe8chB3c38rLGSmPuB4hKvx8AMM8keFJqtxRY"
      }
    },
    "4": {
      "bitcoin": {
        "p2pkh": {
          "address": "mx1AfpDHrLdEjB3PjFTx4xkqRSNgjQSNtb",
          "private_key": "8b453115472f95bc9023e31556c2e14ac354d52b4be357044bd429dda5e1ceb2",
          "public_key": "038372c07f0b424089d541ba89ecca3d0d97e537feaa4ff541c5efb4f6af30e61f",
          "wif": "cSFRbW8HxyXkovovyzzWVT6qR1hrKfWRe5qfr6SnYeuvu1Gc455p"
        },
        "p2tr": {
          "address": "tb1pq3m975h2696sky32xswf0jnz0wfz27azzqsv4dhtg42tzf6vw5ysz6lv64",
          "private_key": "262a6cbc9b6bced94a279f3da2b7e7fbfc6243abc39b4c0faed990caa87ba8fe",
          "public_key": "03d2ed6b485b545312b33eedac68caa92de56f0775802f0dbe7f88718c8cbaa83e",
          "wif": "cNrtcxZvNErcZyLM4UybKrGEFtaQ33tHWFgvFL2Kzuumv2JnmekE"
        },
        "p2wpkh": {
          "address": "tb1qul3z8nacekwsk0u4ud8zzpzrv2pttlqusfhcg6",
          "private_key": "cdf7e8e0fa09fc95dbffd0cd1bd4a5d3ab88ce00de21f9b527d2345635d87133",
          "public_key": "030502fa1b8038346becdf657dcb358ecc306708c62f2aa8f16e5701bea89b1b04",
          "wif": "cUV5SFpgqrmjq6D8LCdfwTGN1tcUKBRNLNfnyzaKdMWGGnVQZrTg"
        }
      },
      "stacks": {
        "address": "ST28ZFAR3H1B9EBZYSS5XGD2851EBHDDVWXV75FJY",
        "private_key": "5bf083343442f835d4ec1c34f7c5d1dd76f52f69078a8e5d516fcb0612abe1c6",
        "public_key": "03d546396a9c1754435aa304813b772101b57bd29b4f1144e7531bad437d5fc70c",
        "wif": "cQfRKCFJuMaxx1gYKrGTrxStX9u9zLmQqD7tZuZ6bfRJskYs4i9f"
      }
    },
    "5": {
      "bitcoin": {
        "p2pkh": {
          "address": "mw3AFcxXeaMKvmJZpNLhBWVbsbaGx7Rchs",
          "private_key": "4d285e89dcb5928a7e3e5b47015176745007e41f0bf35445ff803832085bd7e7",
          "public_key": "03fcbdad50975161371ea411346e0e26c58ef6aef613b72c72dc649c31817312db",
          "wif": "cQAgkLNvbstwNBfLzsFFpW5pfqmmjdAYayF561WJDUtQEiTzF8gG"
        },
        "p2tr": {
          "address": "tb1p5n96emegc39tvdy0mpgj6wtp3yn0yc7kw63qa3vuey0k2zcyemts4ln4cj",
          "private_key": "1df4202ee4ba2f56adb692cba4b8ee869d94cc99cb63933b68fd8cc7a4cf04b8",
          "public_key": "021bdcc420a49bb4147e7e5765eaa4cf2ff8559a061160972af7ae4c4312e77b50",
          "wif": "cNavkYwg9yeC8GhoeyjWUPUJ7siiva9XedoxfZ9Y9VKd8qvp1PuN"
        },
        "p2wpkh": {
          "address": "tb1q94pn9xuwwz4l0lmtz9anx2jl7fuma3mxepclxs",
          "private_key": "faf13ce8b462477e926c3e96c0b0a2e7123a820c109bf2d734322ce11610023e",
          "public_key": "03e514666bdae7da116c32769588df95beb89d8947b6addbf121b87db809591930",
          "wif": "cVzVzmqNUoJmmsDgQ9uMmyTVvoKBm1evXoRALYosfrDd9AnZg4kD"
        }
      },
      "stacks": {
        "address": "ST3VGNBNE5EDXHXJVFWACV99YEJHS6DTBEAGC8CA0",
        "private_key": "72cda716fe055549285a9f35899b80e3193f795927e14dcf4ade3bc3b7a97e6b",
        "public_key": "02e2afce7e29aabdbc885cd7664c6b1121f831e1974fa2722ca15f2edf0c762f74",
        "wif": "cRRs5odoRRU3visB9KRVW5b3TuQEx3DTyVUYs5gaH5SYi76yACLg"
      }
    },
    "6": {
      "bitcoin": {
        "p2pkh": {
          "address": "mtNgX9W3GaP1d7DhxG7TZETF1arnRN7NL8",
          "private_key": "831b2ddc34f030e8fd57a2451ca6bb9efe727791148a42ae3ee33bbd9e629de0",
          "public_key": "03c2c034477bd1f65354de66ccd276cd5070919a5a3e8dd9481469507c55394100",
          "wif": "cRyZ8wGZZV3T6BFr6MVMGDRUiZsRA6eR7Qd5Y7FHEAoXSoCDsSjF"
        },
        "p2tr": {
          "address": "tb1p653ex7pm29zsy9jc4qqfjnzfnxac5f7caj2tr823swesk9djq07qj4jsc9",
          "private_key": "8725c2a80012eff7bce03449a983405b64f6ab5c9994fec62fece56c61023220",
          "public_key": "02b979fe1be9309b38f53100816950b25c0d63a2dfca81a731903ece7d5bae7b7c",
          "wif": "cS7QmvdqjDxmfypEsVxv4WWaipdDzEx7HdnJkoPMYuhiWT7Nq2M6"
        },
        "p2wpkh": {
          "address": "tb1qfakjd3dw2j3j07aez5uhhyvnwyke2edr2h4ccs",
          "private_key": "168f272244eb6aca28d2f9c59fa194e334d2e12f1f37059f248d63c35c69009f",
          "public_key": "032129638899ae9d3311260cf3afea5098e642ed7fb657a3c88278cdee8af6ab9e",
          "wif": "cNLZ56oEKddQ6qLM43BBvJuboSf8AiV2wUsJ4iywMQjBA6Z3CcqT"
        }
      },
      "stacks": {
        "address": "ST2JK0F1EMCT9277NQNBMERZ458X4J3MQ1C693SG4",
        "private_key": "1395f12df64aa54de9594690b32846182cefdc8b9166e28f131c53f66f9083a4",
        "public_key": "02cca48f1204326750bbc9e9a41fcaa019c98ff4be468a6c0a8a54df42bd514e32",
        "wif": "cNEmpzaegw5rwJJYCxvPVQXb9GVNuXRoRwN87FRyJHPaXom2msyF"
      }
    },
    "7": {
      "bitcoin": {
        "p2pkh": {
          "address": "mfcD3Spbtqq1uHJCVetRY3N8d5Zp929RUk",
          "private_key": "19e378c6b75bf9306b3d174909a5b0f84cc502126b925b8d462c449d90487861",
          "public_key": "02739808fa770282511033cde24c7b06b7f1b9ce57cff918c47de9ff9038be4e37",
          "wif": "cNT2SSgHem3H3ejf73kgLuUuQS4RfGuuoJLzsNBQM1H3YyoNqC2f"
        },
        "p2tr": {
          "address": "tb1p8ppdq8vtyvf5v2p9rw3pw9casrcq6sdg0ty0h0x5ukw97a9xrtasq75vhq",
          "private_key": "54d42839973c88178396e072d6c5c1c90ea43b02755c52123bec0ded25982ef2",
          "public_key": "03727f9d6eeaffd2b67383965e89f70dae4daf8e370f49440102422680ef9ea1ce",
          "wif": "cQRbcfCyyxz3y4zkR8EahqNCFvnhNTKeiAiDoJEMKLaH2KCrfTDf"
        },
        "p2wpkh": {
          "address": "tb1qzkmtlefl3s53utwvq26ucfpcwx0s3m0jluh4ah",
          "private_key": "1657cf437588b6f093beba1cccf1af84e22c2b3b9a63cb20ea40267f3c83e87d",
          "public_key": "022124268150f2bf8236785dda3a0ebfa667b04b99c7add2b7795ef34fa062a875",
          "wif": "cNL8hS188ubNgPLd35M3K6BxiM2LmFB9rW81WAKtDt2apbtdbGe3"
        }
      },
      "stacks": {
        "address": "ST4CYC8F29HW2RQY5AF7Q3M3Q1A55FFS9ZV0QM06",
        "private_key": "3f8020dabb107a602f7bea7c2f4d57c86227573df141274e0b2db2acf8bdf894",
        "public_key": "02c9afd6b492466853deb9591fe6830b658905faa5da08b3ab0410268c9e1f6947",
        "wif": "cPi8yYyu9bhiMrT9qw4D5CcrDLZbbBJULTqugnaeDyVzuXoWnXbL"
      }
    },
    "8": {
      "bitcoin": {
        "p2pkh": {
          "address": "mk6YjfA82GsSnf8fbbQonysRjE5nnFs7wK",
          "private_key": "596706190c702ac710cedccf70cc48c6b86903cda0d57427bdc2e207dbbe9890",
          "public_key": "03772c51a8037e441aa0d6341f4f425366ce6fdfbf8dda21af081df20ea0404d9f",
          "wif": "cQaVGsrqqoNRdf8j7FbKsdh2QdUJ8NEBBLpUQE5XHWGfRWQKeF3m"
        },
        "p2tr": {
          "address": "tb1pkku94htpwftv780u04he3t5w8g7gk8thsxy9sj97jafg9nhfargsdr7e0e",
          "private_key": "b2acdbd18e244d604ab87aebdfbf84ee73e4c6d256d6e9d77d7c6991de0454e5",
          "public_key": "02a172dd29653a1f459f1229080ee0a59f165c64afee3cf6a3b79a512066729cc5",
          "wif": "cTa2HR4tx4y32tPZ6JFYX8JTmo8VzTM29g5WSNgZCixMZmqPSktL"
        },
        "p2wpkh": {
          "address": "tb1q3djynryy5y50wjvpmkmvmenrzxg70p8r3hvdus",
          "private_key": "da0cfc72db41886e5b450f4935ffef77aae11911d3be79fd101c97508b5d9df8",
          "public_key": "034fd1a35ba62b2ada6ec510e6817f5100e403ddf3e5c2dd782c190590b5d051d1",
          "wif": "cUtZek33bYDAXVHCwnvst4btSiN4GHR1UvgZXzKnYKtHw2E3eNn5"
        }
      },
      "stacks": {
        "address": "ST1NP5WMQPJH5VMH2EW3SRBJ46BPZB6ZYJPNFA76J",
        "private_key": "5be428e25948f0ce50edceeee1f7d51595534808ec00967ae9d702a978e9dd7c",
        "public_key": "02e7c79ef9b10435a5c2cd310bfe0c12edd8258d012b1224a48bca2ab735bda6a3",
        "wif": "cQfKsfAEeWhXaFisiCDZYGrMGS5WtU9sqQBK7ubA3yqxdDkK9XMY"
      }
    },
    "9": {
      "bitcoin": {
        "p2pkh": {
          "address": "n1FdWnzgcucGBz4qtyFAPV6Km2SoiEnats",
          "private_key": "c72027b77096efa56ea27c380bc14ad0054d0a5db7922d955e4f1f6df7e3b332",
          "public_key": "0213a67e5795d8b4916ae4026ed7c74cd51a33ad548d891d06022dca05d11495d7",
          "wif": "cUFmx39EzNEk7PLrDqzaKJX7LBMaCiSsDsKP2JWX8DnqwgX12kUX"
        },
        "p2tr": {
          "address": "tb1pp76ysaaf6j2qrpjcqek8hx4tgcwwc6ndp8fvxgavtl95skfm2sps58c9wm",
          "private_key": "c0c5d4f9300ec17cd425e83c9bffeb267053914196356b1321de9c1f741c8b36",
          "public_key": "03f293d166c94dc1d61ee8b9362bab717f2d2e75629af160315de6c04af0e70c60",
          "wif": "cU3Rho19rWo82zTGUs8ZMDmYtStqCto9BX6oVwpDN2DJ7kvG3GXA"
        },
        "p2wpkh": {
          "address": "tb1q59hfq0cpydpdtvwdcj6mgt7nxxwka069zfd4ma",
          "private_key": "37719d1c9279cfc3344dddb81a89b7903bb506aad83db876f674f8fc147d829d",
          "public_key": "02070dc8addc8dd6df68dd81189bee1aa8caac61cee32520eecf0ba2b190d67470",
          "wif": "cPSUdPFTU95LMo2JjVNGdkCZzw7woSJjLSiqGoDP2qmmnuRg1zUa"
        }
      },
      "stacks": {
        "address": "ST7NMR0C8YJ5B0A9CY6WSWEFMJAMG25WCVBATC14",
        "private_key": "4b1683d1d910112f66a3a06b5ea2638eb8fd645b2cbf2dace793687858b67b47",
        "public_key": "021e39c87c12f5527f29989a693fd5c7ab4a1bbc2d4a543d0330316f546328de13",
        "wif": "cQ6fPv6FNTCGWzSeWw7NqLBSNyekmfVDtXPJN1qV1PYm51KsoNbo"
      }
    }
  },
  "mnemonic": "mimic click mind able general beef inflict trade spike sing upset page hunt crowd interest armed sister ring track cram expand vicious brain fun",
  "private_key": "729d29b877d166e68df2a04cd5bc556347decf5f832af99a9d8802326d34e4f0",
  "wif": "cRRVjCYtVEaML9N7GP8M3F4LQgjAGEcqPMMheAaUL8cNv8z3781c"
}
```

</details>

### CODE REVIEW

**Comment by radicleart** [comment]
Amazing turnaround thank you @stjepangolemac !

**Comment by radicleart** [comment]
Check for this is that these two commands generate the same set of keys

```
$ sbtc generate-from new > creds.json
$ npx stx get_stacks_wallet_key "..mnenomic from sbtc command.." "m/44'/5757'/0'/0/0" -t
```

for account 0 ad subsequent accounts (increment the trailing 0).

**Comment by radicleart** [comment]
Comparing sbtc-cli

```
{
  "mnemonic": "business destroy two energy joke address sponsor ivory tenant syrup middle expose number abuse ranch margin blouse patch cherry forget car sad silver rocket",
  "seed_private_key": "c26147a2ea65bb7e3226b43bfb1d3fa08bba8cb518b62e6530560268ebb319aa",
  "credentials": {
    "0": {
      "stacks_private_key": "6e16df46e3ca6e21dfcb69cb50836766d378f9eef4fba0ec85022f5f880f4639",
      "stacks_public_key": "02d1f244f7d308f1940aa66ca802ee12268a6442e2c5984d571b76509173de629d",
      "bitcoin_private_key": "80b1c875517e92c6fe16d3a5f59ff4b3275cdda284f838937063e6b91b875410",
      "bitcoin_public_key": "03b797d9fd175dce49a70480841a41e3062baa55549056409994e644ea576df320",
      "stacks_address": "ST3827Q73RQ8N4GA0RK402ADXNZ206GB5ANTAJN2Y",
      "bitcoin_p2pkh_address": "mfhJM23iVVxoNSXso2z2fpkHBKqc3qMFDN",
      "bitcoin_p2wpkh_address": "tb1qq86wt0et6fxehvj00k3t5dcz45fagtjls9ssk5",
      "bitcoin_taproot_address": "tb1pvnfqmz7mj7zd9a59f6gp4en2k7z0trre4r5x40qjy7rdg08gtd8sgrztps"
    },
    "1": {
```

with stacks-cli

```
$ npx stx get_stacks_wallet_key "business destroy two energy joke address sponsor ivory tenant syrup middle expose number abuse ranch margin blouse patch cherry forget car sad silver rocket" "m/44'/5757'/0'/0/0" -t
[
  {
    "privateKey": "6e16df46e3ca6e21dfcb69cb50836766d378f9eef4fba0ec85022f5f880f463901",
    "publicKey": "02d1f244f7d308f1940aa66ca802ee12268a6442e2c5984d571b76509173de629d",
    "address": "ST3827Q73RQ8N4GA0RK402ADXNZ206GB5ANTAJN2Y",
    "btcAddress": "mzVVokQrfuS7Ww6rDjyqzk2MqKm2ipzruo",
    "wif": "cRGhcDVSoLrPLhweao82kNwZHc4BB3QhXxH6JyJrpJkFmUhr7QRn",
    "index": 0
  }
]```

indicates;

1. stacks_private_key(sbtc-cli) === privateKey(stacks-cli) 
2. stacks_private_key(sbtc-cli) is missing the recovery byte (01) at end of the key.
3. WIF(sbtc-cli) !== WIF(stacks-cli) 
4. bitcoin_p2pkh_address(sbtc-cli) !== btcAddress(stacks-cli)

**Comment by stjepangolemac** [comment]
I've addressed all the issues @radicleart pointed out.

File: sbtc-cli/src/commands/generate.rs [hunk 1]
```diff
@@ -1,143 +1,150 @@
...
-    Ok(PrivateKey::from_wif(wif)?)
+fn value_from_wallet(wallet: &Wallet, credentials_count: usize) -> Value {
+    let mut map = Map::new();
+
+    map.insert("mnemonic".into(), wallet.mnemonic().to_string().into());
+    map.insert(
+        "private_key".into(),
+        hex::encode(wallet.master_key().secret_bytes()).into(),
+    );
+    map.insert("wif".into(), wallet.wif().to_string().into());
```
Comment (friedger) [inline sbtc-cli/src/commands/generate.rs:62]: Is this used at all anywhere? Which wallets?

File: sbtc-cli/src/commands/generate.rs [hunk 2]
```diff
@@ -1,143 +1,150 @@
...
-}
-
-fn private_key_from_wif(wif: &str) -> anyhow::Result<PrivateKey> {
-    Ok(PrivateKey::from_wif(wif)?)
+fn value_from_wallet(wallet: &Wallet, credentials_count: usize) -> Value {
+    let mut map = Map::new();
+
+    map.insert("mnemonic".into(), wallet.mnemonic().to_string().into());
+    map.insert(
+        "private_key".into(),
```
Comment (friedger) [inline sbtc-cli/src/commands/generate.rs:59]: Could be `master_private_key`?

File: stacks-core/src/wallet.rs [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+pub fn bitcoin_derivation_path(
+    network: BitcoinNetwork,
+    kind: BitcoinAddressType,
+    index: u32,
+) -> StacksResult<DerivationPath> {
+    let mut path = "m/".to_string();
+
+    match kind {
+        BitcoinAddressType::P2pkh => path.push_str("44'/"),
+        BitcoinAddressType::P2wpkh => path.push_str("84'/"),
```
Comment (friedger) [inline stacks-core/src/wallet.rs:42]: Do you want to add m/49' ? https://github.com/bitcoin/bips/blob/master/bip-0049.mediawiki

**Review by friedger — APPROVED** [review]
LGTM.

I have some comments that can be improved in the next PR.
I like the structure and abstractions!

**Comment by stjepangolemac** [comment]
I'll create a followup PR with these small fixes.

**Pull request merged**